bugsnag-android-core/src/main/java/com/bugsnag/android/DeliveryHeaders.kt

```diff
@@ -12,19 +16,44 @@ internal const val HEADER_INTERNAL_ERROR = "Bugsnag-Internal-Error"
  *
  * @return the HTTP headers
  */
-internal fun errorApiHeaders(apiKey: String) = mapOf(
-    HEADER_API_PAYLOAD_VERSION to "4.0",
-    HEADER_API_KEY to apiKey,
-    HEADER_BUGSNAG_SENT_AT to DateUtils.toIso8601(Date())
-)
+internal fun errorApiHeaders(payload: EventPayload): Map<String, String?> {
+    return mapOf(
+        HEADER_API_PAYLOAD_VERSION to "4.0",
+        HEADER_API_KEY to payload.apiKey,
+        HEADER_BUGSNAG_SENT_AT to DateUtils.toIso8601(Date()),
+        HEADER_BUGSNAG_INTEGRITY to computeSha1Digest(payload)
+    )
+}
 
 /**
  * Supplies the headers which must be used in any request sent to the Session Tracking API.
  *
  * @return the HTTP headers
  */
-internal fun sessionApiHeaders(apiKey: String) = mapOf(
+internal fun sessionApiHeaders(apiKey: String): Map<String, String?> = mapOf(
     HEADER_API_PAYLOAD_VERSION to "1.0",
     HEADER_API_KEY to apiKey,
     HEADER_BUGSNAG_SENT_AT to DateUtils.toIso8601(Date())
 )
```

Comment:
Does this need `HEADER_BUGSNAG_INTEGRITY` as well?

Reply:
Yes - this was missing from the design, I'll add this in.